ruby/ext/google/protobuf_c/storage.c

```diff
@@ -322,6 +324,43 @@ bool native_slot_eq(upb_fieldtype_t type, void* mem1, void* mem2) {
 }
 
 // -----------------------------------------------------------------------------
+// Map field utilities.
+// -----------------------------------------------------------------------------
+
+bool is_map_field(const upb_fielddef* field) {
+  if (upb_fielddef_label(field) != UPB_LABEL_REPEATED ||
+      upb_fielddef_type(field) != UPB_TYPE_MESSAGE) {
+    return false;
+  }
+  const upb_msgdef* subdef = (const upb_msgdef*)upb_fielddef_subdef(field);
```

Comment:
use upb_downcast_msgdef().

Besides not requiring an explicit cast, it will assert that the cast is correct.